src/errors.rs

```diff
@@ -0,0 +1,63 @@
+use derive_more::Display;
+#[cfg(feature = "std")]
+use derive_more::Error;
+
+macro_rules! impl_error {
+    ($enum:tt, $($variant:tt => $str:expr, )+) => {
+        impl From<$enum> for &'static str {
+            fn from(x: $enum) -> &'static str {
+                match x {
+                    $(
+                        $enum::$variant => { $str },
+                        )*
+                }
+            }
+        }
+    };
+}
+
+#[cfg_attr(feature = "std", derive(Error))]
+#[derive(Debug, Display, PartialEq)]
+pub enum ArithmeticError {
+    #[display(fmt = "overflow")]
+    Overflow,
+    #[display(fmt = "division by zero")]
+    DivisionByZero,
+}
+
+impl_error!(ArithmeticError,
+Overflow => "overflow",
+DivisionByZero => "division by zero",
```

Comment:
I think we should implement `Display` manually and reuse `fn as_str()`

Reply:
done